core/src/test/java/org/elasticsearch/index/mapper/NumberFieldTypeTests.java

```diff
@@ -120,6 +120,20 @@ public void testTermQuery() {
         assertEquals("Cannot search on field [field] since it is not indexed.", e.getMessage());
     }
 
+    public void testRangeQueryWithNegativeBounds() {
+        MappedFieldType ftInt = new NumberFieldMapper.NumberFieldType(NumberType.INTEGER);
+        ftInt.setName("field");
+        ftInt.setIndexOptions(IndexOptions.DOCS);
+        assertEquals(IntPoint.newRangeQuery("field", -3, -2), ftInt.rangeQuery(-3.5, -2.5, true, true, null));
```

Comment:
Sorry I may be a bit confused, but if the range is `[-3.5, -2.5]` then `-2` should not match?